backend/rest_test.go

```diff
@@ -0,0 +1,152 @@
+package backend_test
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/restic/restic/backend"
+	"github.com/restic/restic/backend/rest"
+)
+
+func setupRestBackend(t *testing.T) *rest.Rest {
+	url, _ := url.Parse("http://localhost:8000")
+	backend, _ := rest.Open(url)
+	return backend
+}
+
+func TestRestBackend(t *testing.T) {
+
+	// Initializing a temporary direcory for the rest backend.
+	path, _ := ioutil.TempDir("", "restic-repository-")
+	defer os.RemoveAll(path)
+
+	dirs := []string{
+		path,
+		filepath.Join(path, "data"),
+		filepath.Join(path, "snapshot"),
+		filepath.Join(path, "index"),
+		filepath.Join(path, "lock"),
+		filepath.Join(path, "key"),
+		filepath.Join(path, "temp"),
+	}
+
+	for _, d := range dirs {
+		os.MkdirAll(d, backend.Modes.Dir)
+	}
+
+	// Initialize the router for the repository requests.
+	r := mux.NewRouter()
+
+	// Check if the repository has already been initialized.
```

Comment:
This comment is wrong. How does the code defining a `HandleFunc` check if the repository has been initialized?

For restic, we try to build the code in a way that it is obvious what the code does (e.g. by having good function names), so a comment isn't needed and cannot be wrong if the codes is changed. It takes a while to get used to. Especially comments like in line 45 could be just left out.